edalize/nextpnr.py

```diff
@@ -0,0 +1,168 @@
+# Copyright edalize contributors
+# Licensed under the 2-Clause BSD License, see LICENSE for details.
+# SPDX-License-Identifier: BSD-2-Clause
+
+import logging
+import os.path
+
+from edalize.edatool import Edatool
+from edalize.yosys import Yosys
+from importlib import import_module
+
+logger = logging.getLogger(__name__)
+
+""" Nextpnr backend
+
+This backend provides PnR tool which can be used as standalone
+or subtool in another toolchain.
+
+Tool repository: https://github.com/YosysHQ/nextpnr
+
+"""
+
+
+class Nextpnr(Edatool):
+
+    argtypes = []
+    archs = ["xilinx", "fpga_interchange"]
+    families = ["xc7"]
+
+    @classmethod
+    def get_doc(cls, api_ver):
+        if api_ver == 0:
+            yosys_help = Yosys.get_doc(api_ver)
+            nextpnr_help = {
+                "members" : [
+                    {
+                        "name" : "arch",
+                        "type" : "String",
+                        "desc" : "Target architecture. Legal values are *xilinx*"
+                    },
+                    {
+                        "name" : "output_format",
+                        "type" : "String",
+                        "desc" : "Output file format. Legal values are *fasm*"
```

Comment:
Ok, if I understand correctly, output_format does not really belong in nextpnr because a user can't change the output format. In my eyes, this should be hardcoded in the symbiflow part instead

Reply:
I've removed this option -> `.fasm` is hardcoded in makefile templates